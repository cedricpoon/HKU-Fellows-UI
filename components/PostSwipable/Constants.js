export const PRE_RENDER_NUMBER = 2;
