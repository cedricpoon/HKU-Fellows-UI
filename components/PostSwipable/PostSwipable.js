import React, { Component } from 'react';
import { Tab, Tabs } from 'native-base';
import PropTypes from "prop-types";

import { PostDetails, PostLoadIndicator } from 'hkufui/components';

class PostSwipable extends Component {
  render() {
    const { comments, onRef, solved, ...restProps } = this.props;
    const context = [];

    if (comments) {
      comments.forEach((comment, index) => {
        const { id } = comment;
        context.push(
          <Tab heading={id} key={id}>
            <PostDetails comment={comment} index={index + 1} selectedAnswer={solved === id} />
          </Tab>
        );
      });
    }

    return (
      <Tabs
        renderTabBar={false}
        prerenderingSiblingsNumber={comments ? comments.length : 0}
        ref={onRef}
        {...restProps}
      >
        {comments ? context : (
          <Tab heading='' key=''>
            <PostLoadIndicator page />
          </Tab>
        )}
      </Tabs>
    );
  }
}

PostSwipable.propTypes = {
  comments: PropTypes.array,
  onRef: PropTypes.func,
  solved: PropTypes.string
};

export default PostSwipable;
