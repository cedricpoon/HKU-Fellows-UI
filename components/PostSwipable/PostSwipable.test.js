import React from 'react';
import { shallow } from 'enzyme';
import PostSwipable from './PostSwipable';

describe('Testing PostSwipable Component', () => {

  it('renders with placeholder', () => {
    const wrapper = shallow(
      <PostSwipable />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has onRef', () => {
    const wrapper = shallow(
      <PostSwipable comments={[]} onRef={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('is solved', () => {
    const wrapper = shallow(
      <PostSwipable comments={[]} onRef={() => {}} solved='uid' />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
