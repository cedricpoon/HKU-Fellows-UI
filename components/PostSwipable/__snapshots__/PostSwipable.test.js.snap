// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostSwipable Component has onRef 1`] = `
<ScrollableTabView
  contentProps={Object {}}
  initialPage={0}
  locked={false}
  onChangeTab={[Function]}
  onScroll={[Function]}
  page={-1}
  prerenderingSiblingsNumber={0}
  renderTabBar={false}
  scrollWithoutAnimation={false}
  tabBarPosition="top"
/>
`;

exports[`Testing PostSwipable Component is solved 1`] = `
<ScrollableTabView
  contentProps={Object {}}
  initialPage={0}
  locked={false}
  onChangeTab={[Function]}
  onScroll={[Function]}
  page={-1}
  prerenderingSiblingsNumber={0}
  renderTabBar={false}
  scrollWithoutAnimation={false}
  tabBarPosition="top"
/>
`;

exports[`Testing PostSwipable Component renders with placeholder 1`] = `
<ScrollableTabView
  contentProps={Object {}}
  initialPage={0}
  locked={false}
  onChangeTab={[Function]}
  onScroll={[Function]}
  page={-1}
  prerenderingSiblingsNumber={0}
  renderTabBar={false}
  scrollWithoutAnimation={false}
  tabBarPosition="top"
>
  <Styled(Tab)
    heading=""
    key=""
  >
    <PostLoadIndicator
      page={true}
    />
  </Styled(Tab)>
</ScrollableTabView>
`;
